cli/src/cluster_query.rs

```diff
@@ -1780,8 +1776,39 @@ pub fn process_show_stakes(
 ) -> ProcessResult {
     use crate::stake::build_stake_state;
 
-    let progress_bar = new_spinner_progress_bar();
-    progress_bar.set_message("Fetching stake accounts...");
+    let vote_account_progress_bar = new_spinner_progress_bar();
+    vote_account_progress_bar.set_message("Searching for matching vote accounts...");
+
+    // fetching corresponding vote pubkey from identity pubkey
+    let vote_account_pubkeys = match vote_account_pubkeys {
+        Some(pubkeys) => {
+            let vote_accounts = rpc_client.get_vote_accounts()?;
+
+            let pubkeys: HashSet<String> =
+                pubkeys.iter().map(|pubkey| pubkey.to_string()).collect();
+
+            let vote_account_pubkeys: HashSet<String> = vote_accounts
+                .current
+                .iter()
+                .chain(vote_accounts.delinquent.iter())
+                .filter(|&vote_acc| {
+                    pubkeys.contains(&vote_acc.node_pubkey)
+                        || pubkeys.contains(&vote_acc.vote_pubkey)
+                })
+                .map(|vote_acc| vote_acc.vote_pubkey.to_string())
+                .collect();
+
+            if vote_account_pubkeys.len() != pubkeys.len() {
+                return Err(CliError::RpcRequestError(
+                    "Failed to retrieve matching vote account for pubkey(s).".to_string(),
+                )
+                .into());
+            }
```

Comment:
It would be nice to print exactly which pubkeys could not be found. Could you collect them during the iteration `filter` step above?

Reply:
Now error outputs invalid pubkeys